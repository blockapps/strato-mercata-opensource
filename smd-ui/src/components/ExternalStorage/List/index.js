import React, { Component } from 'react';
import { parseDateFromString } from '../../../lib/dateUtils';
import HexText from '../../HexText';

class List extends Component {

  render() {
    let list = this.props.uploadList.length && this.props.uploadList.map(
      function (list, i) {
        return (
          <tr key={i}>
            <td>
              <small>{list.uri}</small>
            </td>
            <td>
              <small>
                <HexText value={list.contractAddress} classes="smd-pad-2" />
              </small>
            </td>
            <td>
              <small>
                <HexText value={list.hash} classes="smd-pad-2" />
              </small>
            </td>
            <td>
              <small>{parseDateFromString(list.createdAt)}</small>
            </td>
          </tr>
        )
      }
    );

    return (
      <div className="pt-card pt-dark pt-elevation-2">
        <table className="pt-table pt-interactive pt-condensed pt-striped upload-list" style={{ tableLayout: 'fixed', width: '100%' }}>
          <thead>
            <tr>
              <th width="40%"><h5>Document</h5></th>
              <th width="20%"><h5>Address</h5></th>
              <th width="20%"><h5>Hash</h5></th>
              <th width="20%"><h5>Uploaded</h5></th>
            </tr>
          </thead>

          <tbody>
            {list || <tr><td colSpan={4}>No Data</td></tr>}
          </tbody>
        </table>
      </div>
    );
  }
}

export default List;
