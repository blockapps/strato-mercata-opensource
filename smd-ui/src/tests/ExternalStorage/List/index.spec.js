import React from 'react';
import List, { mapStateToProps } from '../../../components/ExternalStorage/List';
import { uploadList } from '../storageMock';

describe('List: index', () => {

  describe('render component', () => {

    test('without values', () => {
      const props = {
        uploadList: []
      };

      const wrapper = shallow(
        <List {...props} />
      );

      expect(wrapper.debug()).toMatchSnapshot();
    });

    test('with values', () => {
      const props = {
        uploadList: uploadList
      };

      const wrapper = shallow(
        <List {...props} />
      );

      expect(wrapper.debug()).toMatchSnapshot();
    });

  });

});