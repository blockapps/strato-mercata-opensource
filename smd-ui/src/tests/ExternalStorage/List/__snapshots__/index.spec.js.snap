// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List: index render component with values 1`] = `
"<div className=\\"pt-card pt-dark pt-elevation-2\\">
  <table className=\\"pt-table pt-interactive pt-condensed pt-striped upload-list\\" style={{...}}>
    <thead>
      <tr>
        <th width=\\"40%\\">
          <h5>
            Document
          </h5>
        </th>
        <th width=\\"20%\\">
          <h5>
            Address
          </h5>
        </th>
        <th width=\\"20%\\">
          <h5>
            Hash
          </h5>
        </th>
        <th width=\\"20%\\">
          <h5>
            Uploaded
          </h5>
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          <small>
            https://strato-external-storage.s3.amazonaws.com/1529905060401-widescreen.jpeg
          </small>
        </td>
        <td>
          <small>
            <HexText value=\\"c918420c68346af5fe2aef067faf7b103afde5ed\\" classes=\\"smd-pad-2\\" />
          </small>
        </td>
        <td>
          <small>
            <HexText value={[undefined]} classes=\\"smd-pad-2\\" />
          </small>
        </td>
        <td>
          <small>
            2018-06-25 12:37:42 AM
          </small>
        </td>
      </tr>
      <tr>
        <td>
          <small>
            https://strato-external-storage.s3.amazonaws.com/1529909829846-jellyfish-25-mbps-hd-hevc.3gp
          </small>
        </td>
        <td>
          <small>
            <HexText value=\\"c0d225c56af10ad58eb2c16bd8dfaca3273894ef\\" classes=\\"smd-pad-2\\" />
          </small>
        </td>
        <td>
          <small>
            <HexText value={[undefined]} classes=\\"smd-pad-2\\" />
          </small>
        </td>
        <td>
          <small>
            2018-06-25 01:57:32 AM
          </small>
        </td>
      </tr>
      <tr>
        <td>
          <small>
            https://strato-external-storage.s3.amazonaws.com/1529912097617-soap-bubble-1958650_960_720.jpg
          </small>
        </td>
        <td>
          <small>
            <HexText value=\\"51a867e95aa68efcd2c320352e0861e4f015af88\\" classes=\\"smd-pad-2\\" />
          </small>
        </td>
        <td>
          <small>
            <HexText value={[undefined]} classes=\\"smd-pad-2\\" />
          </small>
        </td>
        <td>
          <small>
            2018-06-25 02:35:01 AM
          </small>
        </td>
      </tr>
      <tr>
        <td>
          <small>
            https://strato-external-storage.s3.amazonaws.com/1529915329415-widescreen.jpeg
          </small>
        </td>
        <td>
          <small>
            <HexText value=\\"9c22ec56dd721cd3ca138dc1d1a05d567e019e36\\" classes=\\"smd-pad-2\\" />
          </small>
        </td>
        <td>
          <small>
            <HexText value={[undefined]} classes=\\"smd-pad-2\\" />
          </small>
        </td>
        <td>
          <small>
            2018-06-25 03:28:53 AM
          </small>
        </td>
      </tr>
      <tr>
        <td>
          <small>
            https://strato-external-storage.s3.amazonaws.com/1529923413618-SampleVideo_360x240_30mb.mkv
          </small>
        </td>
        <td>
          <small>
            <HexText value=\\"edf7fa43718c27d10e6d49bfa8d8942578cc1a87\\" classes=\\"smd-pad-2\\" />
          </small>
        </td>
        <td>
          <small>
            <HexText value={[undefined]} classes=\\"smd-pad-2\\" />
          </small>
        </td>
        <td>
          <small>
            2018-06-25 05:43:56 AM
          </small>
        </td>
      </tr>
      <tr>
        <td>
          <small>
            https://strato-external-storage.s3.amazonaws.com/1529924931049-widescreen.jpeg
          </small>
        </td>
        <td>
          <small>
            <HexText value=\\"9479054a30b89cdd4a512e127049deb042e7e199\\" classes=\\"smd-pad-2\\" />
          </small>
        </td>
        <td>
          <small>
            <HexText value={[undefined]} classes=\\"smd-pad-2\\" />
          </small>
        </td>
        <td>
          <small>
            2018-06-25 06:08:54 AM
          </small>
        </td>
      </tr>
      <tr>
        <td>
          <small>
            https://strato-external-storage.s3.amazonaws.com/1529925597965-widescreen.jpeg
          </small>
        </td>
        <td>
          <small>
            <HexText value=\\"132b62b6b32660aeb707aaed5bfd4cbaed9eb81b\\" classes=\\"smd-pad-2\\" />
          </small>
        </td>
        <td>
          <small>
            <HexText value={[undefined]} classes=\\"smd-pad-2\\" />
          </small>
        </td>
        <td>
          <small>
            2018-06-25 06:20:00 AM
          </small>
        </td>
      </tr>
      <tr>
        <td>
          <small>
            https://strato-external-storage.s3.amazonaws.com/1529931868271-1529923413618-SampleVideo_360x240_30mb.mkv
          </small>
        </td>
        <td>
          <small>
            <HexText value=\\"57ffd407be86add24fb9ff6125708494b7c09a25\\" classes=\\"smd-pad-2\\" />
          </small>
        </td>
        <td>
          <small>
            <HexText value={[undefined]} classes=\\"smd-pad-2\\" />
          </small>
        </td>
        <td>
          <small>
            2018-06-25 08:04:55 AM
          </small>
        </td>
      </tr>
      <tr>
        <td>
          <small>
            https://strato-external-storage.s3.amazonaws.com/1529934246777-jellyfish-25-mbps-hd-hevc.3gp
          </small>
        </td>
        <td>
          <small>
            <HexText value=\\"2425395e37a3623de0f1f6e87781cb9e39fba179\\" classes=\\"smd-pad-2\\" />
          </small>
        </td>
        <td>
          <small>
            <HexText value={[undefined]} classes=\\"smd-pad-2\\" />
          </small>
        </td>
        <td>
          <small>
            2018-06-25 08:44:27 AM
          </small>
        </td>
      </tr>
      <tr>
        <td>
          <small>
            https://strato-external-storage.s3.amazonaws.com/1529934327270-soap-bubble-1958650_960_720.jpg
          </small>
        </td>
        <td>
          <small>
            <HexText value=\\"b928fe10d9d9ece8755b93dfddc32eb1c6db3fe7\\" classes=\\"smd-pad-2\\" />
          </small>
        </td>
        <td>
          <small>
            <HexText value={[undefined]} classes=\\"smd-pad-2\\" />
          </small>
        </td>
        <td>
          <small>
            2018-06-25 08:45:30 AM
          </small>
        </td>
      </tr>
      <tr>
        <td>
          <small>
            https://strato-external-storage.s3.amazonaws.com/1529999523270-widescreen.jpeg
          </small>
        </td>
        <td>
          <small>
            <HexText value=\\"9d5fdf1738593c2f93f098b55a765485a1574d1f\\" classes=\\"smd-pad-2\\" />
          </small>
        </td>
        <td>
          <small>
            <HexText value={[undefined]} classes=\\"smd-pad-2\\" />
          </small>
        </td>
        <td>
          <small>
            2018-06-26 02:52:06 AM
          </small>
        </td>
      </tr>
    </tbody>
  </table>
</div>"
`;

exports[`List: index render component without values 1`] = `
"<div className=\\"pt-card pt-dark pt-elevation-2\\">
  <table className=\\"pt-table pt-interactive pt-condensed pt-striped upload-list\\" style={{...}}>
    <thead>
      <tr>
        <th width=\\"40%\\">
          <h5>
            Document
          </h5>
        </th>
        <th width=\\"20%\\">
          <h5>
            Address
          </h5>
        </th>
        <th width=\\"20%\\">
          <h5>
            Hash
          </h5>
        </th>
        <th width=\\"20%\\">
          <h5>
            Uploaded
          </h5>
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td colSpan={4}>
          No Data
        </td>
      </tr>
    </tbody>
  </table>
</div>"
`;
